### PR TITLE
feat: asg + lambda - first draft

### PR DESCRIPTION
**Pull request by anichari**

### CODE REVIEW

File: cloudformation-templates/zs_cc_cf_template_asg_gwlb.yaml [hunk 1]
```diff
@@ -530,6 +539,7 @@ Resources:
               - 'autoscaling:DescribeAutoScalingInstances'
               - 'autoscaling:CompleteLifecycleAction'
               - 'autoscaling:RecordLifecycleActionHeartbeat'
+              - 'autoscaling:SetInstanceHealth'
```
Comment (vkrishnamurthy-zscaler) [inline cloudformation-templates/zs_cc_cf_template_asg_gwlb.yaml:542]: please add
                "ec2:DescribeInstances",
                "ec2:DescribeTags",

File: cloudformation-templates/zs_cc_cf_template_asg_gwlb.yaml [hunk 2]
```diff
@@ -667,6 +677,211 @@ Resources:
...
+        Statement:
+          - Effect: Allow
+            Action:
+              - 'autoscaling:DescribeLifecycleHookTypes'
+              - 'autoscaling:DescribeLifecycleHooks'
+              - 'autoscaling:DescribeAutoScalingInstances'
+              - 'autoscaling:CompleteLifecycleAction'
+              - 'autoscaling:RecordLifecycleActionHeartbeat'
+              - 'autoscaling:SetInstanceHealth'
+              - 'ec2:DescribeInstanceStatus'
```
Comment (vkrishnamurthy-zscaler) [inline cloudformation-templates/zs_cc_cf_template_asg_gwlb.yaml:757]: Please add
                "ec2:DescribeInstances",
                "ec2:DescribeTags",

**Review by vkrishnamurthy-zscaler — APPROVED** [review]
Please add additional rules.   describe tags is for next week as I need to know CC VM's from others.  DescribeInstances is what lambda uses and Janus doesn't use it(?).  looks good.  More after deployment

File: cloudformation-templates/zs_cc_cf_template_asg_gwlb.yaml (hunk 1 above)
Comment (anichari) [reply cloudformation-templates/zs_cc_cf_template_asg_gwlb.yaml:542]: are you asking for both asg policy and asg policy for lambda to add these two actions ?

File: cloudformation-templates/zs_cc_cf_template_asg_gwlb.yaml (hunk 2 above)
Comment (anichari) [reply cloudformation-templates/zs_cc_cf_template_asg_gwlb.yaml:757]: are you asking for both asg policy and asg policy for lambda to add these two actions ?

Comment (vkrishnamurthy-zscaler) [reply cloudformation-templates/zs_cc_cf_template_asg_gwlb.yaml:757]: just asg policy for lambda only, please ignore the other

**Review by vkrishnamurthy-zscaler — APPROVED** [review]

File: cloudformation-templates/zs_cc_cf_template_asg_gwlb.yaml [hunk 3]
```diff
@@ -242,6 +243,14 @@ Parameters:
         AllowedPattern: '^(.*|[0-9]+)$'
         ConstraintDescription: >-
           When specified, Auto-scaling Group Warm Pool Minimum Size should be a whole number
+      AsgS3BucketName:
```
Comment (jmolnar-zscaler) [inline cloudformation-templates/zs_cc_cf_template_asg_gwlb.yaml:246]: i'm confused. are we hosting a public S3 bucket with the zip file for all customers to pull from? if so, why do i need a copy in tf github?

File: cloudformation-templates/zs_cc_cf_template_asg_gwlb.yaml [hunk 4]
```diff
@@ -667,6 +677,213 @@ Resources:
...
+            Action:
+              - 'logs:CreateLogStream'
+              - 'logs:PutLogEvents'
+            Resource:
+              - !GetAtt 
+                - CcCloudwatchLogGroupForLambda
+                - Arn
+          - Effect: Allow
+            Action:
+              - 'logs:CreateLogGroup'
```
Comment (jmolnar-zscaler) [inline cloudformation-templates/zs_cc_cf_template_asg_gwlb.yaml:731]: @anichari take a look at this commit. After talking with venkatesh, I ended up dropping the CreateLogGroup IAM in lieu of a separate Log group creation in order to set a custom 14 day retention time.

https://github.com/zscaler/terraform-aws-cloud-connector-modules/commit/ab2e1dd6fdfcda7509b4f114f2e7a2d4f0e8f7f4#diff-ba0f81d84e66eac04b6f0f063d0c824ee174c680c697e8d451c125ab60b83f37

File: cloudformation-templates/zs_cc_cf_template_asg_gwlb.yaml (hunk 3 above)
Comment (anichari) [reply cloudformation-templates/zs_cc_cf_template_asg_gwlb.yaml:246]: No, we are not using a public s3 bucket. The user workflow for CFT is that, they get the template and the zip file when they download the release from github. After that, the user uploads the lambda zip file into an s3 bucket in their org etc and then, when running through the cloudformation parameters, they enter the s3 bucket details for cloudformation lambda resource to use.

**Pull request closed without merging**